packages/kbn-apm-synthtrace/src/cli/utils/start_live_data_upload.ts

```diff
@@ -59,22 +51,46 @@ export async function startLiveDataUpload({
         `Requesting ${new Date(bucketFrom).toISOString()} to ${new Date(bucketTo).toISOString()}`
       );
 
-      const next = logger.perf('execute_scenario', () =>
-        generate({ range: timerange(bucketFrom.getTime(), bucketTo.getTime()) })
-      );
+      const generatorsAndClients = generate({
+        range: timerange(bucketFrom.getTime(), bucketTo.getTime()),
+        client: { logsEsClient, apmEsClient },
+      });
+
+      const generatorsAndClientsArray = castArray(generatorsAndClients);
+
+      const streams = generatorsAndClientsArray.map(({ client }) => {
+        const stream = new PassThrough({ objectMode: true });
+        client.index(stream);
+        return stream;
```

Comment:
one question here: do you need multiple streams? can you not call client.index on the same stream? what happens if you do?

Reply:
Looking at all the clients, none of them implements a custom index function at the moment. This means every client literally calls the same BaseClient.index function. 2 challenges i see
1. In future if any client want to do something custom implementing the index function, they won't be able to if we use a single client
2. The `--clean` flag, the way it works is, every client specifies the dataStream to clean. If we use a single client, it will only clean that dataStream instead of other clients `datastream`